controlplane/pkg/nsm/nsm_heal.go

```diff
@@ -153,27 +219,55 @@ func (srv *networkServiceManager) requestOrClose(logPrefix string, ctx context.C
 		logrus.Infof("%v Heal: Connection recovered: %v", logPrefix, connection)
 	}
 }
-func (srv *networkServiceManager) waitAnyNSE(clientConnection *model.ClientConnection) {
+
+
+func (srv *networkServiceManager) waitSpecificNSE(ctx context.Context, clientConnection *model.ClientConnection) bool {
+	discoveryClient, err := srv.serviceRegistry.DiscoveryClient()
+	if err != nil {
+		logrus.Errorf("Failed to connect to Registry... %v", err)
+		// Still try to recovery
+		return false
+	}
+
 	st := time.Now()
+
+	networkService := clientConnection.Endpoint.NetworkService.Name
```

Comment:
`clientConnection.Endpoint` is a reference. If it'll be nil it will panic

Reply:
Not not logically it could be nil, in this case it will fail earlier.